src/KM_FormMain.pas

```diff
@@ -436,12 +436,21 @@ procedure TFormMain.RGPlayerClick(Sender: TObject);
 
 
 procedure TFormMain.chkSuperSpeedClick(Sender: TObject);
+var Speed: Single;
 begin
   if (gGameApp.Game = nil)
-  or (gGameApp.Game.IsMultiplayer and not MULTIPLAYER_SPEEDUP and not gGameApp.Game.IsReplay) then
+  or (gGameApp.Game.IsMultiplayer
+    and not gGameApp.Game.IsMPGameSpeedUpAllowed
+    and not MULTIPLAYER_SPEEDUP
+    and not gGameApp.Game.IsReplay) then
     Exit;
 
-  gGameApp.Game.SetGameSpeed(IfThen(chkSuperSpeed.Checked, 300, 1), False);
+  if gGameApp.Game.IsMPGameSpeedUpAllowed then
```

Comment:
Ah I see, so you're basically using "IsMPGameSpeedUpAllowed" as "IsMultiplayer". Yes lets rename GetDefaultMPGameSpeed to GetDefaultGameSpeed and make it work with SP and MP to avoid these confusing checks.